lib/fizzy/types.hpp

```diff
@@ -324,9 +324,16 @@ struct Element
     std::vector<FuncIdx> init;
 };
 
-// https://webassembly.github.io/spec/core/binary/modules.html#code-section
+/// The element of the code section.
+/// https://webassembly.github.io/spec/core/binary/modules.html#code-section
+///
+/// TODO: This type has currently 3 slightly different uses. Refactor uses to eliminate
```

Comment:
Well I'd prefer to split it in this PR, it doesn't seem too difficult. We'd have something like
```
struct code_view
{
    bytes_view code_binary;
};
parser_result<code_view> parse(const uint8_t* pos, const uint8_t* end);

Code parse_code(code_view code);
```

Reply:
This is nice suggestion except it limits lazy code parsing.